fe/fe-core/src/test/java/org/apache/doris/nereids/rules/rewrite/PullUpProjectUnderLimitTest.java

```diff
@@ -35,14 +36,16 @@
 
 class PullUpProjectUnderLimitTest implements MemoPatternMatchSupported {
     private final LogicalOlapScan scan1 = PlanConstructor.newLogicalOlapScan(0, "t1", 0);
+    private final LogicalOlapScan scan2 = PlanConstructor.newLogicalOlapScan(1, "t2", 0);
 
     @Test
     void test() {
         List<NamedExpression> exprs = ImmutableList.of(
-                scan1.getOutput().get(0),
+                scan1.getOutput().get(0).alias("id"),
                 new Cast(scan1.getOutput().get(1), VarcharType.SYSTEM_DEFAULT).alias("cast")
         );
         LogicalPlan limit = new LogicalPlanBuilder(scan1)
+                .join(scan2, JoinType.LEFT_OUTER_JOIN, ImmutableList.of())
```

Comment:
Do we need to cover right outer and cross? since the code handle these two kinds.